roles/wireguard/templates/iface_config.conf.j2

```diff
@@ -17,28 +17,42 @@ Table = {{ item.table }}
 
 {% for peer in item.peers %}
 [Peer]
-PublicKey = {{ hostvars[peer].wg_pubkeys[item.iface_name] }}
+PublicKey = {{ hostvars[peer].wg_pubkeys[item.iface_name] | default(wireguard_config_directory[peer][item.iface_name].wg_pubkey) }}
 {% if item.psk_enabled is defined %}
-PreSharedKey = {{ hostvars[peer].pre_shared_keys[item.iface_name] }}
+PreSharedKey = {{ hostvars[peer].pre_shared_keys[item.iface_name] | default(wireguard_config_directory[peer][item.iface_name].pre_sharing_key) }}
 {% endif %}
-AllowedIPs = {{ (hostvars[peer].wg_ifaces
-    | selectattr('iface_name', 'equalto', item.iface_name)
-    | map(attribute='ip_addresses_cidr')
-    | first
-    + (item.allowed_ips | default([])))
-    | list
-    | join(', ')
-}}
-Endpoint = {{
-    [ (hostvars[peer].wg_ifaces
-      | selectattr('iface_name', 'equalto', item.iface_name)
-      | map(attribute='peering_ip')
-      | first),
+AllowedIPs = {{
+    (
       (hostvars[peer].wg_ifaces
       | selectattr('iface_name', 'equalto', item.iface_name)
-      | map(attribute='peering_port')
-      | first)
-    ] | join(':')
+      | map(attribute='ip_addresses_cidr')
+      | first
+      + (item.allowed_ips | default([])))
+      | list
+      | join(', ')
+    ) | default(
+      (wireguard_config_directory[peer][item.iface_name].ip_addresses_cidr
+      + (item.allowed_ips | default([])))
+      | list
+      | join(', ')
```

Comment:
Can the join be applied once, to the final list, after the fallback to wireguard_config_directory? So, `(hostvars | … | default(wiregr)) | join(', ')`.

Reply:
Trying, this *should* work in theory.

Reply:
This change seems to be done in `Endpoint` only (where `join(':')`), but not in `AllowedIPs` (where `join(', ')`).

Reply:
Oooops, sorry. You were right. Fixed.